### ISSUE TITLE
[ENHANCEMENT] Reschedule and set task as failed when some conditions are bad

### ISSUE THREAD

**Issue by Writtscher** [kagkarlsson/db-scheduler#121]
At the moment it is only a failure handler is able to mark a run a failed. So this means if some condition is bad I have to throw a exception so my failure handler can reschedule and mark the execution as failed. 

I can reschedule my task from within a task (because there is access to the schedulerClient: executionContext.schedulerClient) but not mark it as failed. IMO the schedulerClient needs a  "markAsFailedAndReschedule method.

What do you think?

**Comment by Writtscher** [comment]
Btw. soon I will have more spare time I'd like to contribute so I hope you are fine with it. Got some ideas that might be useful.

**Comment by kagkarlsson** [comment]
Yeah I see how that can be useful. Will have to have a closer look on how that could be implemented though.

**Comment by kagkarlsson** [comment]
Contributions are most definitely welcome 👍

**Comment by kagkarlsson** [comment]
Closing this. Upcoming fix in existing PR.
